### PR TITLE
refactor(hooks,util): typescript

### PR DESCRIPTION
**Pull request by adnasa**
#### Summary

- updates eslintrc
- adds tsconfig to `packages/utils`, `packages/hooks`
- converts `packages/hooks` and most of `packages/utils` to typescript.

I'll keep it as Draft, since the build fails.
I welcome help on updating the `eslintrc` for further changes needed there.

### CODE REVIEW

File: packages/hooks/src/use-field-id/use-field-id.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import { getFieldId } from '@commercetools-uikit/utils';
+
+type TSignatureCreatesId = () => string;
```
Comment (adnasa) [inline packages/hooks/src/use-field-id/use-field-id.ts:4]: This should be imported from `get-field-id`

File: packages/hooks/src/use-row-selection/use-row-selection.ts [hunk 2]
```diff
@@ -0,0 +1,152 @@
+import React from 'react';
+
+type TRow = {
+  id: string;
+  [key: string]: any;
```
Comment (adnasa) [inline packages/hooks/src/use-row-selection/use-row-selection.ts:5]: I wanted to add here

```js
[id: if value of TRow.id]: boolean
```

not sure if that is possible

File: packages/hooks/src/use-row-selection/use-row-selection.ts [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+  React.useDebugValue(selectionState);
+
+  return [selectionState, dispatch];
+};
+const useRowSelection = (nameOfFlag: string, rows: TRow[]) => {
+  const [selectionState, dispatch] = useSelectionReducer(rows, nameOfFlag);
+  const selectableRows: TRow[] = rows.map((item: TRow) => ({
+    ...item,
+    // @TODO: add a check that `nameOfFlag` is already a key
+    // defined in rows[], so that we don't accidentally override it.
```
Comment (adnasa) [inline packages/hooks/src/use-row-selection/use-row-selection.ts:119]: out of scope.
Should be fixed in the current implementation.

File: packages/hooks/src/use-sorting/use-sorting.ts [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+  return {
+    items: sortState.items,
+    sortedBy: sortState.sortedBy,
+    sortDirection: sortState.sortDirection,
+    onSortChange,
+  };
+};
+
+export default useSorting;
```
Comment (adnasa) [inline packages/hooks/src/use-sorting/use-sorting.ts:104]: lots of generics here.

**Comment by emmenko** [comment]
Thanks for kick starting this!

I would suggest to finish and merge this first #1671 as it simplifies a lot the TS and bundling setup.

**Comment by adnasa** [comment]
@emmenko 
Good morning. okay, cool. are you going to finish that?

**Comment by emmenko** [comment]
FYI: I opened a PR to migrate the utils since it's one of the most commonly used package internally and it prevents from migrating other packages. https://github.com/commercetools/ui-kit/pull/1732

I suppose we can close this and you can migrate the hooks separately.

**Comment by adnasa** [comment]
moin, yes I saw your PR
Thanks for that!

**Pull request closed without merging**